Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection_test.py

```diff
@@ -0,0 +1,541 @@
+    }
+
+
+def test_get_action_data(mocker):
+    import MicrosoftDefenderAdvancedThreatProtection as atp
+    mocker.patch.object(atp, 'get_machine_action_by_id_request', return_value=ACTION_DATA_API_RESPONSE)
+    res = atp.get_machine_action_data("123456")
+    assert res['ID'] == "123456"
+    assert res['Status'] == "Succeeded"
+
+
+def test_get_machine_investigation_package_command(mocker):
+    import MicrosoftDefenderAdvancedThreatProtection as atp
+    mocker.patch.object(atp, 'get_investigation_package_request', return_value=INVESTIGATION_PACKAGE_API_RESPONSE)
+    mocker.patch.object(atp, 'get_machine_action_data', return_value=INVESTIGATION_ACTION_DATA)
+    _, res, _ = atp.get_machine_investigation_package_command()
+    assert res['MicrosoftATP.MachineAction(val.ID === obj.ID)'] == INVESTIGATION_ACTION_DATA
+
+
+def test_get_investigation_package_sas_uri_command(mocker):
+    import MicrosoftDefenderAdvancedThreatProtection as atp
+    mocker.patch.object(atp, 'get_investigation_package_sas_uri_request', return_value=INVESTIGATION_SAS_URI_API_RES)
+    _, res, _ = atp.get_investigation_package_sas_uri_command()
+    assert res['MicrosoftATP.InvestigationURI(val.Link === obj.Link)'] == {
+        'Link': 'https://userrequests-us.securitycenter.windows.com:443/safedownload/'
+                'WDATP_Investigation_Package.zip?token=test1'}
+
+
+def test_restrict_app_execution_command(mocker):
+    import MicrosoftDefenderAdvancedThreatProtection as atp
+    mocker.patch.object(atp, 'restrict_app_execution_request', return_value=MACHINE_ACTION_API_RESPONSE)
+    mocker.patch.object(atp, 'get_machine_action_data', return_value=MACHINE_ACTION_DATA)
+    _, res, _ = atp.restrict_app_execution_command()
+    assert res['MicrosoftATP.MachineAction(val.ID === obj.ID)'] == MACHINE_ACTION_DATA
+
+
+def test_remove_app_restriction_command(mocker):
+    import MicrosoftDefenderAdvancedThreatProtection as atp
+    mocker.patch.object(atp, 'remove_app_restriction_request', return_value=MACHINE_ACTION_API_RESPONSE)
+    mocker.patch.object(atp, 'get_machine_action_data', return_value=MACHINE_ACTION_DATA)
+    _, res, _ = atp.remove_app_restriction_command()
+    assert res['MicrosoftATP.MachineAction(val.ID === obj.ID)'] == MACHINE_ACTION_DATA
+
+
+def test_stop_and_quarantine_file_command(mocker):
+    import MicrosoftDefenderAdvancedThreatProtection as atp
+    mocker.patch.object(atp, 'stop_and_quarantine_file_request', return_value=MACHINE_ACTION_API_RESPONSE)
+    mocker.patch.object(atp, 'get_machine_action_data', return_value=MACHINE_ACTION_DATA)
+    _, res, _ = atp.stop_and_quarantine_file_command()
+    assert res['MicrosoftATP.MachineAction(val.ID === obj.ID)'] == MACHINE_ACTION_DATA
+
+
+def test_get_investigations_by_id_command(mocker):
+    import MicrosoftDefenderAdvancedThreatProtection as atp
+    mocker.patch.object(demisto, 'args', return_value={'id': '123', 'limit': '1', 'offset': '0'})
```

Comment:
add tests for negative /  non int  limit and offset, here and for every other command